### PR TITLE
Filter 3rtparty directories in coverage

### PR DESCRIPTION
**Pull request by alkino**
And split actions of coverage

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        run:  |
+          mkdir build && cd build
+          cmake .. -DPYTHON_EXECUTABLE=$(which python3) -DCMAKE_C_FLAGS="-coverage -O0" -DCMAKE_CXX_FLAGS="-coverage -O0";
+          make
+          (cd ..;  lcov --capture  --initial --directory . --no-external --output-file build/coverage-base.info)
+          lcov --remove coverage-base.info --output-file coverage-base-filtered.info "*/ext/*"
+          make test
+          (cd ..; lcov --capture  --directory . --no-external --output-file build/coverage-run.info)
+          lcov --add-tracefile coverage-base-filtered.info --add-tracefile coverage-run.info --output-file coverage-combined.info
+          genhtml coverage-combined.info --output-directory coverage --demangle-cpp --ignore-errors source
```
Comment (ohm314) [inline .github/workflows/coverage.yml:53]:
```suggestion
          (cd ..;  lcov --capture  --initial --directory . --no-external --output-file build/coverage-base.info)
          make test
          (cd ..; lcov --capture  --directory . --no-external --output-file build/coverage-run.info)
          lcov --add-tracefile coverage-base.info --add-tracefile coverage-run.info --output-file coverage-combined.info
          lcov --remove coverage-combined.info --output-file coverage-combined-filtered.info "*/ext/*"
          genhtml coverage-combined-filtered.info --output-directory coverage --demangle-cpp --ignore-errors source
```
Should we not filter at the end to get the clean result?

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Build and Test for Coverage
+        id: codecov
+        shell: bash
+        working-directory: ${{runner.workspace}}/nmodl
+        run:  |
+          mkdir build && cd build
+          cmake .. -DPYTHON_EXECUTABLE=$(which python3) -DCMAKE_C_FLAGS="-coverage -O0" -DCMAKE_CXX_FLAGS="-coverage -O0";
+          make
+          (cd ..;  lcov --capture  --initial --directory . --no-external --output-file build/coverage-base.info)
+          lcov --remove coverage-base.info --output-file coverage-base-filtered.info "*/ext/*"
```
Comment (ohm314) [inline .github/workflows/coverage.yml:49]: I wonder, the build directory does not show up, but we have generated files in `build/src/ast/*` do you think they can be explicitly included?

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          make test
+          (cd ..; lcov --capture  --directory . --no-external --output-file build/coverage-run.info)
+          lcov --add-tracefile coverage-base-filtered.info --add-tracefile coverage-run.info --output-file lcov.info
+          bash <(curl -s https://codecov.io/bash) -X gcov
```
Comment (ohm314) [inline .github/workflows/coverage.yml:53]: maybe also delete the intermediate `.info` files to not confuse codecov?

Comment (ohm314) [reply .github/workflows/coverage.yml:53]:
```suggestion
          bash <(curl -s https://codecov.io/bash) -X gcov -f lcov.info
```
have you tried this?

**Review by ohm314 — APPROVED** [review]
looks good

**Review by alexsavulescu — COMMENTED** [review]
LGTM, waiting for codecov comment :)

**Comment by pramodk** [comment]
@alkino : do we expect codecov to have comment in the PR with coverage?

**Comment by alkino** [comment]
Yes but codecov is waiting that ci is finished

**Comment by pramodk** [comment]
Did not understand which CI it's waiting for 🤔 . Anyway, if I should merge this, let me know and I press the button :)

**Pull request merged**